lib/cluster.js

```diff
@@ -27,7 +27,19 @@ function Worker(options) {
   if (options === null || typeof options !== 'object')
     options = {};
 
-  this.suicide = undefined;
+  this.exitedVoluntarily = undefined;
+
+  Object.defineProperty(this, 'suicide', {
+    get: function() {
+      // TODO: Print deprecation message.
+      return this.exitedVoluntarily;
+    },
+    set: function(val) {
+      // TODO: Print deprecation message.
+      this.exitedVoluntarily = val;
+    }
+  });
```

Comment:
This seems read-only anyways, setting it from the outside would be pointless and non-sensical.

For at least LTS I suggest instead doing something like:

```js
this.setVoluntaryExit = function(val) {
  this.exitedVoluntarily = val;
  this.suicide = val;
};
```